### ISSUE TITLE
6 Second lag when capturing with Fuji X-T2

### ISSUE THREAD

**Issue by FlashMuller** [gphoto/gphoto2#229]
**Describe the bug**

When (successfully) shooting with my Fujifilm X-T2 it will lock itself and the application for exactly 6 seconds after the image has been captured. It seems to be waiting for some signal from the cam, which never happens. As this is the case, I would also be happy to just have a hack / workaround of setting the timeout for the specific step to like 10ms. Hopefully all necessary data is attached. Let me know if you need any further information.
Best wishes and Thanks,
Daniel

**Name the camera**
It is the Fujifilm X-T2 on current firmware.

**libgphoto2 and gphoto2 version**
0.001121 main                        (2): gphoto2 2.5.20
0.001180 main                        (2): gphoto2 wurde mit den folgenden Optionen übersetzt:
0.001214 main                        (2):  + gcc (C compiler used)
0.001245 main                        (2):  + popt (mandatory, for handling command-line parameters)
0.001275 main                        (2):  + exif (for displaying EXIF information)
0.001304 main                        (2):  + no cdk (for accessing configuration options)
0.001334 main                        (2):  + no aa (for displaying live previews)
0.001363 main                        (2):  + jpeg (for displaying live previews in JPEG format)
0.001392 main                        (2):  + no readline (for easy navigation in the shell)
0.001430 main                        (2): libgphoto2 2.5.22
0.001464 main                        (2): libgphoto2 wurde mit den folgenden Optionen übersetzt:
0.001494 main                        (2):  + all camlibs
0.001523 main                        (2):  + gcc (C compiler used)
0.001552 main                        (2):  + ltdl (for portable loading of camlibs)
0.001581 main                        (2):  + EXIF (for special handling of EXIF files)
0.001617 main                        (2): libgphoto2_port 0.12.0
0.001651 main                        (2): libgphoto2_port wurde mit den folgenden Optionen übersetzt:
0.001682 main                        (2):  + iolibs: disk ptpip serial usb1 usbdiskdirect usbscsi
0.001712 main                        (2):  + gcc (C compiler used)
0.001741 main                        (2):  + ltdl (for portable loading of iolibs)
0.001770 main                        (2):  + USB (libusb1, for USB cameras)
0.001799 main                        (2):  + serial (for serial cameras)
0.001828 main                        (2):  + no resmgr (serial port access and locking)
0.001857 main                        (2):  + no ttylock (serial port locking)
0.001886 main                        (2):  + no lockdev (serial port locking)
0.001916 main                        (2): CAMLIBS env var not set, using compile-time default instead
0.001945 main                        (2): IOLIBS env var not set, using compile-time default instead
0.001974 main                        (2): invoked with following arguments:
0.002003 main                        (2):   --debug
0.002033 main                        (2):   --debug-logfile=/home/pi/HDD/speedTest.txt
0.002062 main                        (2):   --capture-image

**To Reproduce**
It happens for any shot.

3.563262 gp_port_check_int           (3): Reading 24 = 0x18 bytes from interrupt endpoint...
3.713562 gp_port_set_timeout         (2): Setting port timeout to 20000 milliseconds.
3.713747 camera_wait_for_event       (2): waiting for events timeout 3000 ms
3.713797 gp_port_get_timeout         (2): Current port timeout is 20000 milliseconds.
3.713813 gp_port_set_timeout         (2): Setting port timeout to 3000 milliseconds.
**3.713833 gp_port_check_int           (3): Reading 24 = 0x18 bytes from interrupt endpoint...
6.715597 gp_port_check_int           (3): Reading 24 = 0x18 bytes from interrupt endpoint...
9.719013 ptp_usb_event [usb.c:587]   (0): Reading PTP event failed: Zeitüberschreitung beim Lesen oder Schreiben des Ports (-10)** _in english: Timeout while reading or writing port_
9.719172 gp_port_set_timeout         (2): Setting port timeout to 20000 milliseconds.
9.722093 gp_camera_free              (2): Freeing camera...
9.722174 gp_camera_exit              (2): Exiting camera ('Fuji Fujifilm X-T2')...
9.722212 camera_unprepare_capture    (2): Unprepare_capture

**Comment by FlashMuller** [comment]
Never mind, I found the solution in another issue. To anyone else searching this (without giving any guarantees):
Find the GPhoto2 main.c file (should be in the gphoto2 subfolder of you installation). Find the function generic_capture and at the bottom of that function the parameter waitingtime, which by default is 3000 Milliseconds. I changed it to 50 Milliseconds and it works like a charme. Dont forget to _./configure --> make --> sudo make install_ once you changed the file.
Background, taken from the comments in the file and one developer statement on that issue: GPhoto doesn't know if a jpg file will be written after the raw file has been captured. Therefor it will wait a certain amount of time (in my case twice, therefor not only 3 but 6 seconds delay) and check if a file was written. As I know there won't be a jpg file following I can as well let the timeout happen instantly.
Best wishes,
Daniel
